pkg/cmd/pulumi/new.go

```diff
@@ -1112,3 +1117,29 @@ func containsWhiteSpace(value string) bool {
 	}
 	return false
 }
+
+func validateStackProjectName(b backend.Backend, stackName, projectName string) error {
+	if stackName == "" {
+		return nil
+	}
+	if projectName == "" {
+		return nil
+	}
+
+	if strings.Count(stackName, "/") == 2 {
```

Comment:
A comment here adding context on exactly which case we're detecting here would be helpful.
e.g., if the user supplied `--stack organization/$project/$stack`, verify that the project name matches `--project $project`.